src/CofreeBot/Bot.hs

```diff
@@ -145,6 +150,33 @@ fixBot (Bot b) = go
     go :: s -> Behavior m i o
     go s = Behavior $ \i -> second go <$> b s i
 
+-- TODO: A bi-parser typeclass
+type Serializable a = (Read a, Show a)
+
+fixBotPersistent :: forall m s i o. (MonadIO m, Serializable s) => FilePath -> Bot m s i o -> s -> IO (Behavior m i o)
+fixBotPersistent cachePath (Bot bot) initialState = do
+  saveState cachePath initialState
+  pure go
+  where
+    go :: Behavior m i o
+    go = Behavior $ \i ->
+      liftIO (readState cachePath) >>= \case
+        Nothing -> error "ERROR: Failed to read Bot State from disk."
+        Just oldState -> do
+          (output, newState) <- bot oldState i
+          liftIO $ saveState cachePath newState
+          pure (output, go)
+
+readState :: Read s => String -> IO (Maybe s)
```

Comment:
It would be nice if we could avoid opening and closing this file over and over, but I don't have any good ideas at this moment about how to do it.

Reply:
I guess we could read the file once and then persist it in an IORef/MVar/TVAR for the lifecycle of the bot? If we go that route we should put it in a followup PR